### DISCUSSION TITLE
Supermini ProMicro

### DISCUSSION THREAD

**Discussion by louisvangeldrop** [Moddable-OpenSource/moddable#1442] [Porting]
I just bought a couple of these devices at aliexpress:
https://github.com/joric/nrfmicro/wiki/Alternatives#supermini-nrf52840
I would like to use them with the Moddable software. Any suggestion of how to achieve that is welcome.

**Comment by mkellner** [comment]
Hello!

The first step is to put Moddable Bootloader onto the device. The picture of that device shows no buttons, but you can connect between RST and GND (or power?) to have a reset button.

The Moddable Bootloader uses a double-tap on the reset button to force the device into programming mode, so you need at least the reset button. When the device is configured with the Moddable bootloader and you are running debug builds, then the reset button is almost never necessary.

If you can get the device to mount on your desktop as a drive (it may be named NRF52BOOT or something similar), then you can try to copy the bootloader directly onto the device.

If it does not work, you will have to use JTAG and connect to the programming ports. It looks like there are some pads on the bottom of the device.

https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/devices/nrf52.md#bootloader-1

Once the Moddable bootloader is in place, then using the device is almost identical to other devices.
